packages/notifier/src/subscribe.js

```diff
@@ -0,0 +1,177 @@
+};
+harden(subscribeEach);
+
+/**
+ * @template T
+ * @param {ERef<LatestTopic<T>>} topic
+ * @param {bigint} [localUpdateCount]
+ * @param {IteratorReturnResult<T>} [terminalResult]
+ * @returns {ForkableAsyncIterator<T, T>}
+ */
+const cloneLatestIterator = (topic, localUpdateCount, terminalResult) => {
+  let mutex = Promise.resolve();
+
+  /**
+   * Request the next update record from the topic, updating our local state,
+   * and convert it to an iterator result.
+   *
+   * @returns {Promise<IteratorResult<T, T>>}
+   */
+  const maybeRequestNextResult = async () => {
+    if (terminalResult) {
+      // We've reached the end of the topic, just keep returning the last result
+      // without further requests.
+      return terminalResult;
+    }
+
+    // Send the next request now, skipping past intermediate updates.
+    const { value, updateCount } = await E(topic).getUpdateSince(
+      localUpdateCount,
+    );
+    // Make sure the next request is for a fresher value.
+    localUpdateCount = updateCount;
+
+    // Make an IteratorResult.
+    if (updateCount === undefined) {
+      terminalResult = harden({ done: true, value });
+      return terminalResult;
+    }
+    return harden({ done: false, value });
+  };
+
+  return Far('LatestIterator', {
+    fork: () => cloneLatestIterator(topic, localUpdateCount, terminalResult),
+    next: async () => {
+      // In this adaptor, once `next()` is called and returns an unresolved
+      // promise, further `next()` calls will also return unresolved promises
+      // but each call will not trigger another `topic` request until the prior
+      // one has settled.
+      //
+      // This linear queueing behavior is only needed for code that uses the
+      // async iterator protocol explicitly. When this async iterator is
+      // consumed by a for/await/of loop, `next()` will only be called after the
+      // promise for the previous iteration result has fulfilled. If it fulfills
+      // with `done: true`, the for/await/of loop will never call `next()`
+      // again.
+      //
+      // See
+      // https://2ality.com/2016/10/asynchronous-iteration.html#queuing-next()-invocations
+      // for an explicit use that sends `next()` without waiting.
+
+      if (terminalResult) {
+        // We've reached the end of the topic, just keep returning the last
+        // result.
+        return terminalResult;
+      }
+
+      // BEGIN CRITICAL SECTION - synchronously enqueue and reassign `mutex`
+      //
+      // Use `mutex` to ensure that we have no more than a single request in
+      // flight.
+      const nextResult = mutex.then(maybeRequestNextResult);
+      mutex = nextResult.then(sink, sink);
+      // END CRITICAL SECTION
+
+      return nextResult;
+    },
+  });
+};
+
+/**
+ * @template T
+ * @param {ERef<LatestTopic<T>>} topic
+ * @returns {ForkableAsyncIterator<T, T>}
+ */
+const makeLatestIterator = topic =>
+  cloneLatestIterator(topic, undefined, undefined);
```

Comment:
Clearer without explicitly providing optional arguments?
```suggestion
const makeLatestIterator = topic => cloneLatestIterator(topic);
```

Reply:
I was torn.  I'll follow your guidance.